src/components/EditableDropdown/EditableDropdown.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+import Dropdown from '../Dropdown';
+
+
+const TextBox = styled.input`
+  margin-top: 2px;
+  padding: 6px;
+  font-size: 14px;
+  width: 240px;
+  outline-color: ${props => (props.changed ? 'orange' : 'default')};
+  border: none;
+  z-index: 10;
+  position: absolute;
+  top: 0.75em;
+  left: 0.75em;
+
+  &:focus {
+    outline: none; 
+  }
+`;
+
+class EditableDropdown extends React.Component {
+  constructor(props, context) {
+    super(props, context)
+
+    this.dropdown = null;
+
+    this.state = {
+      currentValue: "",
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.setDropdownRef = this.setDropdownRef.bind(this);
+    this.handleFocus = this.handleFocus.bind(this);
+  }
+
+  setDropdownRef = ref => {
+    this.dropdown = ref;
+  }
+
+  handleChange(ev, value, label) {
+    if (this.props.onChange) {
+      this.props.onChange(ev, value, label);
+    }
+
+    this.setState({
+      currentValue: label,
+    });
+  }
+
+  handleFocus() {
+    if (this.props.onFocus) {
+        this.props.onFocus();
+    }
+    if (this.dropdown) {
+      this.dropdown.handleClick();
+    }
+  }
+
+  handleInputChange(event) {
+    this.setState({
+      currentValue: event.target.label,
+    });
+  }
+
+  render () {
+    return (
+      <div style={{position: "relative"}}>
+        <TextBox type="text" onFocus={this.handleFocus} value={this.state.currentValue} onChange={this.handleInputChange} placeholder={this.props.placeholder} />
+        <Dropdown className={this.props.className} innerRef={this.setDropdownRef} onChange={this.handleChange} items={this.props.items}/>
```

Comment:
Why do we need to save the `Dropdown` ref? I don't see it used, but there may be something I'm not seeing.